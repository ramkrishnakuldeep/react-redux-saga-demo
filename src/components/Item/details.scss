.item {
    background: white;

    .item-img {
        position: relative;
        img {
            object-fit: contain;
            width: 200px;
        }
    }

    .item-detail {
        padding: 10px;
        .item-price{
            display: flex;
            align-items: center;
            justify-content: space-between;
        }
    }
    
    
}