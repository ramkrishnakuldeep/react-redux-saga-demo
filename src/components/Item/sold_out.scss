@import '../../resources/styles/variables';

.sold-out {
    position: absolute;
    left: 0;
    top: 0;
    .triangle {
        border-right: 70px solid transparent;
        border-top: 70px solid $red;
    }
    &>span {
        position: absolute;
        left: 2px;
        top: 15px;
        color: $white;
        transform: rotate(-45deg);
    }
}