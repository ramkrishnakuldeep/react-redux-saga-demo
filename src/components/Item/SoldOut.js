import React from 'react';
import './sold_out.scss';

const SoldOut = () => {
    return (
        <div className="sold-out">
            <div className="triangle"> </div>
            <span> SOLD </span>
        </div>
    );
};

export default SoldOut;
