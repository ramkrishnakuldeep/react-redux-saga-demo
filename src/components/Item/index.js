import React from 'react';
import { Link } from 'react-router-dom';
// import SoldOut from './soldOut';
import './details.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart } from '@fortawesome/free-solid-svg-icons';
import { itemDetails } from '../../propTypes';
import SoldOut from './soldOut';

const Item = ({ item }) => {
    return (
        <React.Fragment>
            <Link to={`/items/${item.id}`} style={{ textDecoration: 'none' }}>
                <div className="item">
                    <div className="item-img">
                        <img src={item.image} />
                        {
                            item.is_sold_out &&
                            <SoldOut />
                            // <div className="sold-out">
                            //     <div className="triangle"> </div>
                            //     <span> SOLD </span>
                            // </div>
                        }
                    </div>
                    <div className="item-detail">
                        <div> {item.name} </div>
                        <div className="item-price">
                            <span> 	&#165; {item.price} </span>
                            <span>
                                <FontAwesomeIcon icon={faHeart} />
                                <span> {item.like_count} </span>
                            </span>
                        </div>
                    </div>
                </div>
            </Link>
        </React.Fragment>
    );
};

Item.propTypes = { item: itemDetails };

export default Item;
