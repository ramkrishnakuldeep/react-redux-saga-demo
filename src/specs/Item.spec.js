import React from 'react';
import { shallow } from '../enzyme';
import Item from '../components/Item';
import { MockItem, MockSoldOutItem } from './__mock__/data';

describe('Item component', () => {
    it('render soldout item', () => {
        const component = shallow(<Item item={MockSoldOutItem} />);
        expect(component).toMatchSnapshot();
    });

    it('render item', () => {
        const component = shallow(<Item item={MockItem} />);
        expect(component).toMatchSnapshot();
    });
});
