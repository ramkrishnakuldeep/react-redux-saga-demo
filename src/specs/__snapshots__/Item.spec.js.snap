// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component render item 1`] = `
<Fragment>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/items/1"
  >
    <div
      className="item"
    >
      <div
        className="item-img"
      >
        <img
          src="http://dummyimage.com/400x400/000/fff?text=men1"
        />
      </div>
      <div
        className="item-detail"
      >
        <div>
           
          men1
           
        </div>
        <div
          className="item-price"
        >
          <span>
              ¥ 
            51
             
          </span>
          <span>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f004",
                    "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
                  ],
                  "iconName": "heart",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <span>
               
              91
               
            </span>
          </span>
        </div>
      </div>
    </div>
  </Link>
</Fragment>
`;

exports[`Item component render soldout item 1`] = `
<Fragment>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/items/1"
  >
    <div
      className="item"
    >
      <div
        className="item-img"
      >
        <img
          src="http://dummyimage.com/400x400/000/fff?text=men1"
        />
        <SoldOut />
      </div>
      <div
        className="item-detail"
      >
        <div>
           
          men1
           
        </div>
        <div
          className="item-price"
        >
          <span>
              ¥ 
            51
             
          </span>
          <span>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f004",
                    "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
                  ],
                  "iconName": "heart",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <span>
               
              91
               
            </span>
          </span>
        </div>
      </div>
    </div>
  </Link>
</Fragment>
`;
